examples/bigquery_user_info_updater_tool/bigquery_user_info_updater/updater_tools/query_creator.py

```diff
@@ -0,0 +1,336 @@
+# Copyright 2019 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import print_function
+import datetime
+import json
+import logging
+import os
+
+
+class QueryCreator(object):
+    """Class for created customized queries based off of a schema.
+
+    Attributes:
+        schema_path(str): Path to the user tables schema in JSON format.
+        user_id_field_name(str): Name of the field that identifies unique users.
+        ingest_timestamp_field_name(str): Name of the timestamp field that marks
+            the ingestion.
+        project_id(str): Id of project holding BQ resources.
+        dataset_id(str): Id of dataset holding Identity BQ user tables.
+        updates_table_id(str): The ID of the historical table that holds
+            all rows and updates for all users.
+        temp_updates_table_id(str): ID of the intermediary temp table.
+        final_table_id(str): The ID of table that holds one up-to-date row
+            per user.
+
+
+    """
+    def __init__(
```

Comment:
jaketf on Aug 27, 2019 Member
again just a PSA, check out data classes in python3.7